File: functions/generate-pdf/src/lib/html.js
import Handlebars from 'handlebars';
import fs from 'fs';
import { fileURLToPath } from 'url';
import path from 'path';

// Get the directory name properly in ESM
const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// Function to read and compile a template
async function generate(templateName, data) {
    const templatePath = path.resolve(__dirname, `../pages/${templateName}.hbs`);
    const templateSource = await fs.promises.readFile(templatePath, 'utf8');
    const template = Handlebars.compile(templateSource);

    // Apply the data to the template
    const html = template(data);

    return html;
}

export default {
    generate
}

File: functions/generate-pdf/src/lib/html.py
import pybars
import os

async def generate(template_name, data):
    """
    Generate HTML estimate from template and data
    
    Args:
        data (dict): Data to apply to the template
        
    Returns:
        str: Rendered HTML content
    """
    template_path = os.path.join(os.path.dirname(__file__), "../../templates/{}.hbs".format(template_name))
    # Create a Handlebars compiler
    compiler = pybars.Compiler()
    
    # Read the template file
    with open(template_path, 'r') as file:
        template_source = file.read()
    
    # Compile the template
    template = compiler.compile(template_source)
    
    # Apply the data to the template
    return template(data)